### PR TITLE
Introduce `ObjectRef`

### PR DESCRIPTION
**Pull request by omus**
Introduces the `ObjectRef` Julia object which is the Julian interface for dealing with object references. This is similar to how Python handles things. Having this struct allows us to hide the internals of using C++ `ObjectID`s from users.

Addresses part of #77. Extracted from #67

### CODE REVIEW

File: Ray.jl/test/object_ref.jl [hunk 1]
```diff
@@ -0,0 +1,20 @@
+@testset "ObjectRef" begin
+    @testset "basic" begin
+        hex_str = "f" ^ (2 * 28)
+        obj_ref = ObjectRef(hex_str)
+        @test obj_ref == ObjectRef(hex_str)
+        @test string(obj_ref) == hex_str
+    end
+
+    @testset "show" begin
+        hex_str = "f" ^ (2 * 28)
```
Comment (kleinschmidt) [inline Ray.jl/test/object_ref.jl:10]: F

File: Ray.jl/src/object_store.jl [hunk 2]
```diff
@@ -20,6 +22,7 @@ if run in an `@async` task.
 If the task that generated the `ObjectID` failed with a Julia exception, the
 captured exception will be thrown on `get`.
 """
+get(obj_ref::ObjectRef) = _get(take!(ray_jll.get(convert(ray_jll.ObjectID, obj_ref.oid))))
```
Comment (kleinschmidt) [inline Ray.jl/src/object_store.jl:25]: what's the point of the `convert` in there?  should be an `ObjectIDAllocated <: ObjectID` already right?

File: Ray.jl/src/object_ref.jl [hunk 3]
```diff
@@ -0,0 +1,14 @@
+struct ObjectRef
+    oid::ray_jll.ObjectIDAllocated
+end
+
+ObjectRef(hex_str::AbstractString) = ObjectRef(ray_jll.FromHex(ray_jll.ObjectID, hex_str))
+Base.string(obj_ref::ObjectRef) = String(ray_jll.Hex(obj_ref.oid))
```
Comment (kleinschmidt) [inline Ray.jl/src/object_ref.jl:6]: I think this should be something else; the julia docs say that `Base.string` and `Base.show` should be consistent, and they aren't in this case.

Comment (kleinschmidt) [reply Ray.jl/src/object_ref.jl:6]: ah actually I misread, its' _print_ that should be consistent with `string`.  I still dont' think we want to be `print`ing these as a string of hex digits without any additional disambiguation though

File: Ray.jl/src/object_store.jl (hunk 2 above)
Comment (omus) [reply Ray.jl/src/object_store.jl:25]: Whoops, I made a mistake here. This was intended to be:
```suggestion
get(obj_ref::ObjectRef) = _get(take!(ray_jll.get(convert(ray_jll.ObjectID, obj_ref))))
```
Felt a bit odd doing the field access here but either should work. What's your preference?

Comment (kleinschmidt) [reply Ray.jl/src/object_store.jl:25]: I think field access is fine!  This is defining the translation layer so it's fine IMO to access the internal fields, right?

Comment (kleinschmidt) [reply Ray.jl/src/object_store.jl:25]: the `_get(take!(get` turducken is a little funny but I understand the reason for the different levels of indirection

File: Ray.jl/src/object_ref.jl (hunk 3 above)
Comment (omus) [reply Ray.jl/src/object_ref.jl:6]: I'll create an accessor called `hex_identifier` instead

**Review by kleinschmidt — APPROVED** [review]

**Pull request merged**